### ISSUE TITLE
API Change from Kernel Explainer to Tree Explainer?

### ISSUE THREAD

**Issue by andymancodes** [shap/shap#160]
Hi, 

I applied SHAP on my data and it worked fine with:

explainer = shap.KernelExplainer(reg.predict, X_train)
shap_values = explainer.shap_values(X_test, nsamples=100)

but now, since my model is a Gradient Boosted Decision tree ensemble, I want to use the Tree explained since it is better for Trees (haven't read the paper though :( ).

Now if I use,

TreeExplainer instead of Kernel explainer it fails.

How do I use the Tree Explained in this case? My data is just X-train and X_test that worked with the Kernerl explainer.

Thanks for the great lib,
Cheers!

**Comment by slundberg** [comment]
You don't need to pass X_train since TreeExplainer infers the training data from the sample counts stored in the trees.
